### PR TITLE
Fix selectReportsToFillBatch

### PR DESCRIPTION
**Pull request by jarnaud**

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/ccipexec/ocr2.go [hunk 1]
```diff
@@ -1023,16 +1023,17 @@ func (r *ExecutionReportingPlugin) ensurePriceRegistrySynchronization(ctx contex
 // because it picks messages and execution states based on the report[0].Interval.Min - report[len-1].Interval.Max range.
 // Having unexpiredReports not sorted properly will lead to fetching more messages and execution states to the memory than the messagesLimit provided.
 // However, logs from LogPoller are returned ordered by (block_number, log_index), so it should preserve the order of Interval.Min.
-// Single CommitRoot can have up to 256 messages, with current MessagesIterationStep of 800, it means processing 4 CommitRoots at once.
+// Single CommitRoot can have up to 256 messages, with current MessagesIterationStep of 1024, it means processing 4 CommitRoots at once.
 func selectReportsToFillBatch(unexpiredReports []cciptypes.CommitStoreReport, messagesLimit uint64) ([]cciptypes.CommitStoreReport, int) {
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/ccipexec/ocr2.go:1027]: Can we have a couple of test scenarios for these new edge conditions?

File: core/services/ocr2/plugins/ccip/ccipexec/ocr2.go [hunk 2]
```diff
@@ -1023,16 +1023,17 @@ func (r *ExecutionReportingPlugin) ensurePriceRegistrySynchronization(ctx contex
...
 	for index = range unexpiredReports {
-		currentNumberOfMessages += unexpiredReports[index].Interval.Max - unexpiredReports[index].Interval.Min + 1
-		if currentNumberOfMessages >= messagesLimit {
+		reportMsgCount := unexpiredReports[index].Interval.Max - unexpiredReports[index].Interval.Min + 1
+		if currentNumberOfMessages+reportMsgCount > messagesLimit {
 			break
 		}
+		currentNumberOfMessages += reportMsgCount
 	}
 	index = min(index+1, len(unexpiredReports))
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/ccipexec/ocr2.go]: This is wrong right now, no? We should pick `min(index, len(reports))`, shouldn't we?

**Review by mateusz-sekara — APPROVED** [review]

File: core/services/ocr2/plugins/ccip/ccipexec/ocr2.go (hunk 2 above)
Comment (jarnaud) [reply core/services/ocr2/plugins/ccip/ccipexec/ocr2.go]: In fact this is unnecessary now
